### PR TITLE
Eto state conditions in investment

### PR DESCRIPTION
**Pull request by trival**
sets more conditions for wallet visibility and selectability during investment. 

Main logic update is in app/modules/investment-flow/sagas.ts, function getActiveInvestmentTypes.
The rest is mostly updating an Enum name.

### CODE REVIEW

**Comment by trival** [comment]
https://github.com/Neufund/platform-frontend/pull/1654 needs to be merged first

**Comment by trival** [comment]
also handles https://github.com/neufund/platform-frontend/issues/1401

File: app/modules/investment-flow/sagas.ts [hunk 1]
```diff
@@ -178,6 +187,45 @@ function* setGasPrice(): any {
...
+  const etoId = selectInvestmentEtoId(state.investmentFlow);
+  const eto = selectEtoWithCompanyAndContractById(state, etoId);
+
+  let activeTypes: EInvestmentType[] = [
+    EInvestmentType.InvestmentWallet,
+    EInvestmentType.BankTransfer,
+  ];
+
+  // no bank transfer 3 days befor eto end
+  const etoEndDate = eto && eto.contract && eto.contract.startOfStates[EETOStateOnChain.Signing];
```
Comment (rudolfix) [inline app/modules/investment-flow/sagas.ts]: I know that it sounds like splitting hairs but:
1. of current `timedState` == `Whitelist` hide bank transfer 3 days before public phase
2. condition above should be used if `timedState` is public.

That will prevent also people on whitelist from sending money that will be received when we are in public already

File: app/modules/public-etos/selectors.ts [hunk 2]
```diff
@@ -65,3 +65,21 @@ export const selectPublicEtos = (state: IAppState): TEtoWithCompanyAndContract[]
...
+  return contracts && contracts.timedState;
+};
+
+export const selectEtoOnChainStateById = (
+  state: IPublicEtoState,
+  etoId: string,
+): EETOStateOnChain | undefined => {
+  const code = selectEtoPreviewCode(state, etoId);
+  if (code) {
+    return state.contracts[code] && state.contracts[code].timedState;
```
Comment (rudolfix) [inline app/modules/public-etos/selectors.ts]: are you sure that internal state is by `code`, not `etoId`? I think using `code` is better btw. so it's ok, just making sure

Comment (trival) [reply app/modules/public-etos/selectors.ts]: yes, contracts are stored by previewCode, so all should be fine

File: app/modules/investment-flow/sagas.ts (hunk 1 above)
Comment (trival) [reply app/modules/investment-flow/sagas.ts]: 👍

**Pull request merged**